Panic when Gas Output Workers >1, lease = 0, and heights not defined

### Describe the bug:
Panic when gas output workers are enabled, have leasing disabled, and do not define height work range. Params and resulting panic included.

### Steps to Reproduce:
<summary>
Visor startup params
<details>

```
Release "sentinel-monitoring-prod" has been upgraded. Happy Helming!
NAME: sentinel-monitoring-prod
LAST DEPLOYED: Tue Nov 10 12:02:25 2020
NAMESPACE: ntwk-mainnet-fullnode
STATUS: pending-upgrade
REVISION: 10
TEST SUITE: None
HOOKS:
MANIFEST:
---
# Source: sentinel/charts/telegraf/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: sentinel-monitoring-prod-telegraf
  labels:
    helm.sh/chart: telegraf-1.7.23
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: telegraf
    app.kubernetes.io/instance: sentinel-monitoring-prod
---
# Source: sentinel/charts/telegraf/templates/configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: sentinel-monitoring-prod-telegraf
  labels:
    helm.sh/chart: telegraf-1.7.23
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: telegraf
    app.kubernetes.io/instance: sentinel-monitoring-prod
data:
  telegraf.conf: |+
    
    [agent]
      collection_jitter = "0s"
      debug = false
      flush_interval = "10s"
      flush_jitter = "0s"
      hostname = "$HOSTNAME"
      interval = "10s"
      logfile = ""
      metric_batch_size = 1000
      metric_buffer_limit = 10000
      omit_hostname = false
      precision = ""
      quiet = false
      round_interval = true
    [[processors.lotus_peer_id]]
      lotus_api_multiaddr = "${LOTUS_API_MULTIADDR}"
      lotus_api_token = "${LOTUS_API_TOKEN}"
    
    
    [[outputs.postgresql]]
    
    [[outputs.health]]
      service_address = "http://:8888"
      [[outputs.health.compares]]
        field = "buffer_size"
        lt = 5000.0
      [[outputs.health.contains]]
        field = "buffer_size"
    [[inputs.prometheus]]
      namedrop = [
        "lotus_info"
      ]
      urls = [
        "http://fullnode-2-lotus-daemon.ntwk-mainnet-fullnode.svc.cluster.local:1234/debug/metrics"
      ]
    [[inputs.lotus]]
      lotus_api_multiaddr = "${LOTUS_API_MULTIADDR}"
      lotus_api_token = "${LOTUS_API_TOKEN}"
    [[inputs.internal]]
      collect_memstats = false
---
# Source: sentinel/charts/telegraf/templates/role.yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: Role
metadata:
  name: sentinel-monitoring-prod-telegraf
  namespace: ntwk-mainnet-fullnode
  labels:
    helm.sh/chart: telegraf-1.7.23
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: telegraf
    app.kubernetes.io/instance: sentinel-monitoring-prod
rules:
  []
---
# Source: sentinel/charts/telegraf/templates/rolebinding.yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: RoleBinding
metadata:
  name: sentinel-monitoring-prod-telegraf
  labels:
    helm.sh/chart: telegraf-1.7.23
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: telegraf
    app.kubernetes.io/instance: sentinel-monitoring-prod
subjects:
  - kind: ServiceAccount
    name: sentinel-monitoring-prod-telegraf
    namespace: ntwk-mainnet-fullnode
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: sentinel-monitoring-prod-telegraf
---
# Source: sentinel/charts/telegraf/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: sentinel-monitoring-prod-telegraf
  labels:
    helm.sh/chart: telegraf-1.7.23
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: telegraf
    app.kubernetes.io/instance: sentinel-monitoring-prod
spec:
  type: ClusterIP
  ports:
  - port: 8888
    targetPort: 8888
    name: "health"
  selector:
    app.kubernetes.io/name: telegraf
    app.kubernetes.io/instance: sentinel-monitoring-prod
---
# Source: sentinel/charts/visor/templates/service-metrics.yaml
apiVersion: v1
kind: Service
metadata:
  name: sentinel-monitoring-prod-visor-metrics
  labels:
    app: visor
    release: sentinel-monitoring-prod
  annotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "9991"
    prometheus.io/path: /metrics
spec:
  type: ClusterIP
  selector:
      app: visor
      release: sentinel-monitoring-prod
  ports:
    - protocol: TCP
      port: 9991
      targetPort: 9991
      name: metrics
---
# Source: sentinel/charts/telegraf/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: sentinel-monitoring-prod-telegraf
  labels:
    helm.sh/chart: telegraf-1.7.23
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: telegraf
    app.kubernetes.io/instance: sentinel-monitoring-prod
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: telegraf
      app.kubernetes.io/instance: sentinel-monitoring-prod
  template:
    metadata:
      labels:
        app.kubernetes.io/name: telegraf
        app.kubernetes.io/instance: sentinel-monitoring-prod
      annotations:
        checksum/config: b1dc85db20a03b9cdfffa6a00f367f2852d7bf5d49bcd77969f92969569e3ced
    spec:
      serviceAccountName: sentinel-monitoring-prod-telegraf
      containers:
      - name: telegraf
        image: "filecoin/sentinel-drone:v0.4.0-rc1"
        imagePullPolicy: "IfNotPresent"
        resources:
          {}
        env:
        - name: LOTUS_API_MULTIADDR
          value: /dns/fullnode-2-lotus-daemon.ntwk-mainnet-fullnode.svc.cluster.local/tcp/1234
        - name: LOTUS_API_TOKEN
          valueFrom:
            secretKeyRef:
              key: jwt-ro-privs-token
              name: fullnode-2-jwt-secrets
        - name: PGPASSWORD
          valueFrom:
            secretKeyRef:
              key: password
              name: sentinel-monitoring-postgresql-secrets
        - name: PGUSER
          valueFrom:
            secretKeyRef:
              key: user
              name: sentinel-monitoring-postgresql-secrets
        - name: PGHOST
          valueFrom:
            secretKeyRef:
              key: host
              name: sentinel-monitoring-postgresql-secrets
        - name: PGPORT
          valueFrom:
            secretKeyRef:
              key: port
              name: sentinel-monitoring-postgresql-secrets
        - name: PGDATABASE
          valueFrom:
            secretKeyRef:
              key: database
              name: sentinel-monitoring-postgresql-secrets
        volumeMounts:
        - name: config
          mountPath: /etc/telegraf
      volumes:
      - name: config
        configMap:
          name: sentinel-monitoring-prod-telegraf
---
# Source: sentinel/charts/visor/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: sentinel-monitoring-prod-visor
  labels:
    app: visor
    suite: sentinel
spec:
  replicas: 1
  selector:
    matchLabels:
      app: visor
      release: sentinel-monitoring-prod
  template:
    metadata:
      labels:
        app: visor
        suite: sentinel
        chart: visor-2.0.6
        heritage: Helm
        release: sentinel-monitoring-prod
    spec:
      initContainers:
      - name: setup-postgresql
        image: "postgres:12"
        env:
          - name: PGPASSWORD
            valueFrom:
              secretKeyRef:
                name: sentinel-monitoring-postgresql-secrets
                key: password
          - name: PGUSER
            valueFrom:
              secretKeyRef:
                name: sentinel-monitoring-postgresql-secrets
                key: user
          - name: PGHOST
            valueFrom:
              secretKeyRef:
                name: sentinel-monitoring-postgresql-secrets
                key: host
          - name: PGPORT
            valueFrom:
              secretKeyRef:
                name: sentinel-monitoring-postgresql-secrets
                key: port
          - name: PGDATABASE
            valueFrom:
              secretKeyRef:
                name: sentinel-monitoring-postgresql-secrets
                key: database
          - name: POSTGRES_DB
            value: sentinel-monitoring
        command: ["/bin/bash", "-c"]
        args: ["echo \"SELECT 'CREATE DATABASE $(POSTGRES_DB)' WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = '$(POSTGRES_DB)'); \\gexec\" | psql --set=sslmode=require"]
      containers:
      - name: visor
        image: "filecoin/sentinel-visor:v0.3.0-rc1"
        imagePullPolicy: "Always"
        args: ["--tracing=false", "run"]
        ports:
          - containerPort: 9991
            name: metrics
        env:
          - name: LOTUS_API_MULTIADDR
            value: "/dns/fullnode-2-lotus-daemon.ntwk-mainnet-fullnode.svc.cluster.local/tcp/1234"
          - name: LOTUS_API_TOKEN
            valueFrom:
              secretKeyRef:
                name: fullnode-2-jwt-secrets
                key: jwt-ro-privs-token
          - name: FULLNODE_API_INFO
            value: "$(LOTUS_API_TOKEN):$(LOTUS_API_MULTIADDR)"
          - name: PGPASSWORD
            valueFrom:
              secretKeyRef:
                name: sentinel-monitoring-postgresql-secrets
                key: password
          - name: PGUSER
            valueFrom:
              secretKeyRef:
                name: postgresql-secrets
                name: sentinel-monitoring-postgresql-secrets
                key: user
          - name: PGHOST
            valueFrom:
              secretKeyRef:
                name: sentinel-monitoring-postgresql-secrets
                key: host
          - name: PGPORT
            valueFrom:
              secretKeyRef:
                name: sentinel-monitoring-postgresql-secrets
                key: port
          - name: PGDATABASE
            value: sentinel-monitoring
          - name: GOLOG_LOG_LEVEL
            value: "info"
          - name: LOTUS_DB
            value: "postgres://$(PGUSER):$(PGPASSWORD)@$(PGHOST):$(PGPORT)/$(PGDATABASE)?sslmode=require"
          - name: LOTUS_DB_POOL_SIZE
            value: "100"
          - name: VISOR_ALLOW_SCHEMA_MIGRATION
            value: "true"

          - name: VISOR_INDEXHEAD
            value: "true"
          - name: VISOR_INDEXHEAD_CONFIDENCE
            value: "0"
          - name: VISOR_INDEXHISTORY
            value: "false"
          - name: VISOR_STATECHANGE_WORKERS
            value: "1"
          - name: VISOR_STATECHANGE_BATCH
            value: "1"
          - name: VISOR_STATECHANGE_LEASE
            value: "15m"
          - name: VISOR_ACTORSTATE_WORKERS
            value: "1"
          - name: VISOR_ACTORSTATE_BATCH
            value: "100"
          - name: VISOR_ACTORSTATE_LEASE
            value: "30m"
          - name: VISOR_MESSAGE_WORKERS
            value: "1"
          - name: VISOR_MESSAGE_BATCH
            value: "1"
          - name: VISOR_MESSAGE_LEASE
            value: "30m"
          - name: VISOR_GASOUTPUTS_WORKERS
            value: "1"
          - name: VISOR_GASOUTPUTS_BATCH
            value: "100"
          - name: VISOR_GASOUTPUTS_LEASE
            value: "0"
        resources:
          limits:
            cpu: 8000m
            memory: 24Gi
          requests:
            cpu: 4000m
            memory: 12Gi
---
# Source: sentinel/charts/visor/templates/service-monitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: sentinel-monitoring-prod-visor
  namespace: ntwk-mainnet-fullnode
spec:
  namespaceSelector:
    matchNames:
      - ntwk-mainnet-fullnode
  selector:
    matchLabels:
      app: visor
      release: sentinel-monitoring-prod
  endpoints:
  - port: metrics
    path: /metrics
    interval: 30s

NOTES:
.  _________              __  .__              .__
 /   _____/ ____   _____/  |_|__| ____   ____ |  |
 \_____  \_/ __ \ /    \   __\  |/    \_/ __ \|  |
 /        \  ___/|   |  \  | |  |   |  \  ___/|  |__
/_______  /\___  >___|  /__| |__|___|  /\___  >____/
        \/     \/     \/             \/     \/

A Filecoin Network Monitoring and Analysis System
```

</details>
</summary>

Resulting panic:
```
[11:48:47] $ kubectl -n ntwk-mainnet-fullnode logs -f sentinel-monitoring-prod-visor-784f94d47d-fjxfw
2020-11-10T16:48:54.915Z        INFO    visor   commands/run.go:238     Visor version:v0.3.0+rc1-dirty
panic: can't subdivide a range of -9223372036854775808 into 1 pieces
goroutine 1 [running]:
github.com/filecoin-project/sentinel-visor/commands.heightRange.divide(0x0, 0x7fffffffffffffff, 0x1, 0x1, 0x1, 0xc000128120)
        /go/src/github.com/filecoin-project/sentinel-visor/commands/run.go:534 +0x1e9
github.com/filecoin-project/sentinel-visor/commands.glob..func3(0xc0004d82c0, 0x0, 0x0)
        /go/src/github.com/filecoin-project/sentinel-visor/commands/run.go:366 +0x20bf
github.com/urfave/cli/v2.(*Command).Run(0x3a64660, 0xc0003a86c0, 0x0, 0x0)
        /go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:164 +0x4ed
github.com/urfave/cli/v2.(*App).RunContext(0xc000602a80, 0x2928560, 0xc00005e170, 0xc00004e090, 0x3, 0x3, 0x0, 0x0)
        /go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:306 +0x81f
github.com/urfave/cli/v2.(*App).Run(...)
        /go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:215
main.main()
        /go/src/github.com/filecoin-project/sentinel-visor/main.go:115 +0xe30
```
### Visor Version: v0.3.0+rc1

--- a/commands/run.go
+++ b/commands/run.go
@@ -529,15 +529,21 @@ type heightRange struct {
 
 // divide divides a range into n equal subranges. The last range may be undersized.
 func (h heightRange) divide(n int) []heightRange {
-	size := h.max - h.min + 1 // +1 since the range is inclusive
+	if n <= 1 {
+		return []heightRange{h}
+	}
+	size := h.max - h.min
+	if size != math.MaxInt64 {
+		size++ // +1 since the range is inclusive
+	}
 	if int64(n) > size {
 		panic(fmt.Sprintf("can't subdivide a range of %d into %d pieces", size, n))
 	}
-	step := size / int64(n)
+	step := size/int64(n) - 1
 
 	subs := make([]heightRange, n)
 	subs[0].min = h.min
-	subs[0].max = h.min + step - 1
+	subs[0].max = h.min + step
 
 	for i := 1; i < n; i++ {
 		subs[i].min = subs[i-1].max + 1

--- a/commands/run_test.go
+++ b/commands/run_test.go
@@ -1,0 +1,80 @@
+package commands
+
+import (
+	"math"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHeightRangeDivide(t *testing.T) {
+	testCases := []struct {
+		hr   heightRange
+		n    int
+		want []heightRange
+	}{
+		{
+			hr: heightRange{min: 1, max: 10},
+			n:  2,
+			want: []heightRange{
+				{min: 1, max: 5},
+				{min: 6, max: 10},
+			},
+		},
+		{
+			hr: heightRange{min: 1, max: 3},
+			n:  3,
+			want: []heightRange{
+				{min: 1, max: 1},
+				{min: 2, max: 2},
+				{min: 3, max: 3},
+			},
+		},
+		{
+			hr: heightRange{min: 1, max: 4},
+			n:  3,
+			want: []heightRange{
+				{min: 1, max: 1},
+				{min: 2, max: 2},
+				{min: 3, max: 4},
+			},
+		},
+		{
+			hr: heightRange{min: 1, max: 4},
+			n:  2,
+			want: []heightRange{
+				{min: 1, max: 2},
+				{min: 3, max: 4},
+			},
+		},
+		{
+			hr: heightRange{min: 1, max: 4},
+			n:  1,
+			want: []heightRange{
+				{min: 1, max: 4},
+			},
+		},
+		{
+			hr: heightRange{min: 0, max: math.MaxInt64},
+			n:  1,
+			want: []heightRange{
+				{min: 0, max: math.MaxInt64},
+			},
+		},
+		{
+			hr: heightRange{min: 0, max: math.MaxInt64},
+			n:  2,
+			want: []heightRange{
+				{min: 0, max: 4611686018427387902},
+				{min: 4611686018427387903, max: math.MaxInt64},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run("", func(t *testing.T) {
+			got := tc.hr.divide(tc.n)
+			assert.Equal(t, tc.want, got)
+		})
+	}
+}

